ansible_base/resource_registry/tasks/sync.py

```diff
@@ -114,6 +139,185 @@ def fetch_manifest(
+            team_data = team_resp.json()
+            for assignment in team_data.get('results', []):
+                # Handle both object_id and object_ansible_id
+                ansible_id_or_pk = assignment.get('object_ansible_id') or assignment.get('object_id')
+                assignments.add(
+                    AssignmentTuple(
+                        actor_ansible_id=assignment['team_ansible_id'],
+                        ansible_id_or_pk=ansible_id_or_pk,
+                        role_definition_name=assignment['role_definition'],
+                        assignment_type='team',
+                    )
+                )
+    except Exception as e:
+        logger.warning(f"Failed to fetch remote team assignments: {e}")
+
+    return assignments
+
+
+def get_local_assignments() -> set[AssignmentTuple]:
+    """Get local assignments and convert to tuples."""
+    assignments = set()
+
+    # Get user assignments
+    for assignment in RoleUserAssignment.objects.select_related('user', 'role_definition').all():
+        try:
+            user_resource = Resource.get_resource_for_object(assignment.user)
+        except Resource.DoesNotExist:
+            # Skip assignments where the user doesn't have a resource
+            continue
+
+        user_ansible_id = user_resource.ansible_id
+        # Handle both object-scoped and global assignments
+        object_id = assignment.object_id
+        ansible_id_or_pk = None
+        if object_id and assignment.content_type:
+            ansible_id_or_pk = get_ansible_id_or_pk(assignment)
+
+        assignments.add(
+            AssignmentTuple(
+                actor_ansible_id=str(user_ansible_id),
+                ansible_id_or_pk=ansible_id_or_pk if ansible_id_or_pk else None,
+                role_definition_name=assignment.role_definition.name,
+                assignment_type='user',
+            )
+        )
+
+    # Get team assignments
+    for assignment in RoleTeamAssignment.objects.select_related('team', 'role_definition').all():
+        try:
+            team_resource = Resource.get_resource_for_object(assignment.team)
+        except Resource.DoesNotExist:
+            # Skip assignments where the user doesn't have a resource
+            continue
+        team_ansible_id = team_resource.ansible_id
+
+        # Handle both object-scoped and global assignments
+        object_id = assignment.object_id
+        ansible_id_or_pk = None
+        if object_id and assignment.content_type:
+            # For object-scoped assignments, try to get the object's ansible_id
+            ansible_id_or_pk = get_ansible_id_or_pk(assignment)
+
+        assignments.add(
+            AssignmentTuple(
+                actor_ansible_id=str(team_ansible_id),
+                ansible_id_or_pk=ansible_id_or_pk if ansible_id_or_pk else None,
+                role_definition_name=assignment.role_definition.name,
+                assignment_type='team',
+            )
+        )
+
+    return assignments
```

Comment:
For memory use an isolation I think it would clearly be _better_ do do the set comparisons for user & team assignments separately, so you are not building a set in memory of both combined. But I realize this probably isn't going to happen before merge of this.